src/main/java/org/chainoptim/desktop/features/product/controller/ProductsController.java

```diff
@@ -12,53 +12,67 @@
 import javafx.scene.layout.VBox;
 import org.chainoptim.desktop.MainApplication;
 import org.chainoptim.desktop.core.context.TenantContext;
+import org.chainoptim.desktop.core.main.controller.HeaderController;
 import org.chainoptim.desktop.core.main.service.CurrentSelectionService;
 import org.chainoptim.desktop.core.main.service.NavigationServiceImpl;
 import org.chainoptim.desktop.core.user.model.User;
 import org.chainoptim.desktop.features.product.model.Product;
 import org.chainoptim.desktop.features.product.repository.ProductRepository;
 import org.chainoptim.desktop.shared.fallback.FallbackManager;
+import org.chainoptim.desktop.shared.search.model.PaginatedResults;
+import org.chainoptim.desktop.shared.search.model.SearchParams;
 
 import java.io.IOException;
 import java.net.URL;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Optional;
-import java.util.ResourceBundle;
+import java.util.*;
+
 
 public class ProductsController implements Initializable {
 
     private final ProductRepository productRepository;
     private final FallbackManager fallbackManager;
-
+    private final Map<String, String> filtersMap = Map.of(
```

Comment:
Dont put filtersMap (sortOptions) and SearchParams in the middle of the services. Put them right before the constructor.